client/gutenberg/extensions/map/component.js

```diff
@@ -0,0 +1,338 @@
+	};
+	deleteActiveMarker = () => {
+		const { points } = this.props;
+		const { activeMarker } = this.state;
+		const { index } = activeMarker.props;
+		const newPoints = points.slice( 0 );
+
+		newPoints.splice( index, 1 );
+		this.props.onSetPoints( newPoints );
+		this.setState( { activeMarker: null } );
+	};
+	// Various map functions
+	sizeMap = () => {
+		const { map } = this.state;
+		const mapEl = this.mapRef.current;
+		const blockWidth = mapEl.offsetWidth;
+		const maxHeight = window.innerHeight * 0.8;
+		const blockHeight = Math.min( blockWidth * ( 3 / 4 ), maxHeight );
+		mapEl.style.height = blockHeight + 'px';
+		map.resize();
+		this.setBoundsByMarkers();
+	};
+	setBoundsByMarkers = () => {
+		const { zoom, points, onSetZoom } = this.props;
+		const { map, activeMarker, mapboxgl, zoomControl, fit_to_bounds } = this.state;
+
+		const bounds = new mapboxgl.LngLatBounds();
+		if ( ! map || ! points.length || activeMarker ) {
+			return;
+		}
+		points.forEach( point => {
+			bounds.extend( [ point.coordinates.longitude, point.coordinates.latitude ] );
+		} );
+		map.setCenter( bounds.getCenter() );
+
+		// If there are multiple points, zoom is determined by the area they cover,
+		// and zoom control is removed.
+		if ( points.length > 1 ) {
+			map.fitBounds( bounds, {
+				padding: {
+					top: 40,
+					bottom: 40,
+					left: 20,
+					right: 20,
+				},
+			} );
+			this.setState( { fit_to_bounds: true } );
+			map.removeControl( zoomControl );
+			return;
+		}
+		/* Case where points go from multiple to just one. Set zoom to an arbitrarily high level. */
+		if ( fit_to_bounds ) {
+			const newZoom = 12;
+			map.setZoom( newZoom );
+			onSetZoom( newZoom );
+		} else {
+			// If there are one (or zero) points, user can set zoom
+			map.setZoom( parseInt( zoom, 10 ) );
+		}
+		this.setState( { fit_to_bounds: false } );
+		map.addControl( zoomControl );
+	};
+	getMapStyle() {
+		const { map_style, map_details } = this.props;
+		return mapboxMapFormatter( map_style, map_details );
+	}
+	getMapType() {
+		const { map_style } = this.props;
+		switch ( map_style ) {
+			case 'satellite':
+				return 'HYBRID';
+			case 'terrain':
+				return 'TERRAIN';
+			case 'black_and_white':
+			default:
+				return 'ROADMAP';
+		}
+	}
+	// Script loading, browser geolocation
+	scriptsLoaded = () => {
+		const { map_center, points } = this.props;
+		this.setState( { loaded: true } );
+
+		// If the map has any points, skip geolocation and use what we have.
+		if ( points.length > 0 ) {
+			this.initMap( map_center );
+			return;
+		}
+		this.initMap( map_center );
+	};
+	loadMapLibraries() {
+		const { api_key } = this.props;
+		asyncLoader(
+			[
+				{
+					type: 'js',
+					url: 'https://api.tiles.mapbox.com/mapbox-gl-js/v0.51.0/mapbox-gl.js',
+				},
+				{
+					type: 'css',
+					url: 'https://api.tiles.mapbox.com/mapbox-gl-js/v0.51.0/mapbox-gl.css',
+				},
+			],
+			() => {
+				window.mapboxgl.accessToken = api_key;
```

Comment:
We should probably check if `window` and exists (this could be server side or mobile rendered) — any change `window.mapboxgl` wouldn't exist?

Reply:
This one is N/A now, because I'm using webpack dynamic importing, added here: https://github.com/Automattic/wp-calypso/pull/28498/commits/774f983606e3bab33fb0363d725abf658d10092f. Because of this change, the Mapbox library is no longer scoped to `window`.